Source/Orts.Simulation/Simulation/RollingStocks/SubSystems/PowerTransmissions/Axle.cs

```diff
@@ -829,7 +954,69 @@ void StationaryCalculation(float elapsedClockSeconds)
         /// <param name="timeSpan"></param>
         public virtual void Update(float timeSpan)
         {
-            forceToAccelerationFactor = WheelRadiusM * WheelRadiusM / totalInertiaKgm2;
+            // Test to determine whether to use Polach or Pacha adhesion
+                        
+            // Switches between Polach (high performance) adhesion model and Pacha (low performance) adhesion model depending upon the PC performance
+            if(timeSpan < 0.025) // timespan 0.025 = 40 fps screen rate, low timeSpan and high FPS
+            {
+                UsePolachAdhesion = true;
+            }
+            else if(timeSpan > 0.033) // timespan 0.033 = 30 fps screen rate, high timeSpan and low FPS
+            {
+                UsePolachAdhesion = false;
+                if (TrainSpeedMpS > 0 )
+                {
+                    var ScreenFrameRate = 1 / timeSpan;
+                    Trace.TraceInformation("Advanced adhesion model switched to low performance option due to low frame rate {0} at ElapsedClockSeconds of {1}", ScreenFrameRate, timeSpan);
+
+                }
+
+                // Set values for Pacha adhesion
+                WheelSlipThresholdMpS = MpS.FromKpH(AdhesionK / AdhesionLimit);
+                WheelAdhesion = 0.99f;
```

Comment:
Why are we setting this to 99% instead of using the same logic?

Reply:
>
> Why are we setting this to 99% instead of using the same logic?
>
Both of these values were not displayed in the Pacha model. Wheel Adhesion could be displayed, but MaxWheelAdhesion was not calculated for the Pacha model.

Perhaps the more elegant way would be to change the HuD so that these lines are removed when switched to the Pacha mode.

Reply:
But you can calculate MaxWheelAdhesion for the Pacha model too. It may be good having it for debugging. Never mind, I'll refactor this later to make almost everything in axle.cs model independent.